core/src/main/java/hudson/util/RobustReflectionConverter.java

```diff
@@ -290,6 +290,53 @@ public Object unmarshal(final HierarchicalStreamReader reader, final Unmarshalli
         return serializationMethodInvoker.callReadResolve(result);
     }
 
+    private void processAttribute(final Object result, final HierarchicalStreamReader reader, final SeenFields seenFields, String attrAlias) {
+        String attrName = mapper.attributeForAlias(attrAlias);
+        Class classDefiningField = determineWhichClassDefinesField(reader);
+        boolean fieldExistsInClass = fieldDefinedInClass(result, attrName);
+        if (fieldExistsInClass) {
+            Field field = reflectionProvider.getField(result.getClass(), attrName);
+            SingleValueConverter converter = mapper.getConverterFromAttribute(field.getDeclaringClass(), attrName, field.getType());
+            Class type = field.getType();
+            if (converter == null) {
+                converter = mapper.getConverterFromItemType(type);
+                // TODO add fieldName & definedIn args
```

Comment:
Best to preserve the formatting in this type of refactoring so that code comparison is easier.
```suggestion
                converter = mapper.getConverterFromItemType(type); // TODO add fieldName & definedIn args
```